public/app/plugins/datasource/grafana/components/AnnotationQueryEditor.tsx

```diff
@@ -66,49 +90,86 @@ export default function AnnotationQueryEditor({ query, onChange }: Props) {
+        };
+      }
+    } else {
+      delete newQuery.timeRegion;
+    }
+
+    onChange(newQuery);
+  };
+  const onTimeRegionChange = (timeRegion?: TimeRegionConfig) => {
+    onChange({
+      ...annotationQuery,
+      timeRegion,
+    });
+  };
+
   return (
     <FieldSet className={styles.container}>
-      <Field label="Filter by">
+      <Field label="Query type">
         <Select
-          inputId="grafana-annotations__filter-by"
-          options={annotationTypes}
-          value={type}
-          onChange={onFilterByChange}
+          inputId="grafana-annotations__query-type"
+          options={queryTypes}
+          value={grafanaQueryType}
+          onChange={onQueryTypeChange}
         />
       </Field>
-      <Field label="Max limit">
-        <Select
-          inputId="grafana-annotations__limit"
-          width={16}
-          options={limitOptions}
-          value={limit}
-          onChange={onMaxLimitChange}
-        />
-      </Field>
-      {type === GrafanaAnnotationType.Tags && (
+      {grafanaQueryType === GrafanaQueryType.Annotations && (
         <>
-          <Field label="Match any" description={matchTooltipContent}>
-            <Switch id="grafana-annotations__match-any" value={matchAny} onChange={onMatchAnyChange} />
+          <Field label="Filter by">
+            <Select
+              inputId="grafana-annotations__filter-by"
+              options={annotationTypes}
+              value={type}
+              onChange={onFilterByChange}
+            />
           </Field>
-          <Field label="Tags" description={tagsTooltipContent}>
-            <TagFilter
-              allowCustomValue
-              inputId="grafana-annotations__tags"
-              onChange={onTagsChange}
-              tagOptions={getAnnotationTags}
-              tags={tags ?? []}
+          <Field label="Max limit">
+            <Select
+              inputId="grafana-annotations__limit"
+              width={16}
+              options={limitOptions}
+              value={limit}
+              onChange={onMaxLimitChange}
             />
           </Field>
+          {type === GrafanaAnnotationType.Tags && (
+            <>
+              <Field label="Match any" description={matchTooltipContent}>
+                <Switch id="grafana-annotations__match-any" value={matchAny} onChange={onMatchAnyChange} />
+              </Field>
+              <Field label="Tags" description={tagsTooltipContent}>
+                <TagFilter
+                  allowCustomValue
+                  inputId="grafana-annotations__tags"
+                  onChange={onTagsChange}
+                  tagOptions={getAnnotationTags}
+                  tags={tags ?? []}
+                />
+              </Field>
+            </>
+          )}
         </>
       )}
+      {grafanaQueryType === GrafanaQueryType.TimeRegions && (
+        <TimeRegionEditor value={annotationQuery.timeRegion!} onChange={onTimeRegionChange} />
```

Comment:
Are we confident that `timeRegion` will always be defined? Maybe a more defensive coding practice would be to also check that `timeRegion` is defined as part of conditional render logic (so we can drop the `!`)

Reply:
I can add a check just for peace of mind, but if the query type is TimeRegions, I'm pretty sure `timeRegions` will exist.